api/v1alpha3/awscluster_webhook.go

```diff
@@ -80,10 +85,37 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 		)
 	}
 
+	allErrs = append(allErrs, r.validateAllowedCIDRBlocks()...)
+
 	return aggregateObjErrors(r.GroupVersionKind().GroupKind(), r.Name, allErrs)
 }
 
+func (r *AWSCluster) validateAllowedCIDRBlocks() []*field.Error {
+	var errs field.ErrorList
+
+	if r.Spec.Bastion.DisableIngressRules && len(r.Spec.Bastion.AllowedCIDRBlocks) > 0 {
+		errs = append(errs,
+			field.Forbidden(field.NewPath("spec", "bastion", "allowedCIDRBlocks"), "cannot be set if spec.bastion.disableIngressRules is true"),
+		)
+		return errs
+	}
+
+	for _, cidr := range r.Spec.Bastion.AllowedCIDRBlocks {
+		if _, _, err := net.ParseCIDR(cidr); err != nil {
+			errs = append(errs,
+				field.Invalid(field.NewPath("spec", "bastion", "allowedCIDRBlocks"), cidr, "must be a valid CIDR block"),
```

Comment:
```suggestion
				field.Invalid(field.NewPath("spec", "bastion", fmt.Sprintf("allowedCIDRBlocks[%d]", i)), cidr, "must be a valid CIDR block"),
```

I think we're going to want the array index included in the field path

Reply:
thanks @ncdc !  I didn't realize the Field Path could or should include the array index. Updated!